### DISCUSSION TITLE
Rollback & Restorable Dashboards: Provide better UX in "An unexpected error happened" window: if given JSON is incorrect

### DISCUSSION THREAD

**Discussion by Dentrax** [grafana/grafana#45674] [General]
I accidentally pushed an incorrect JSON config in the UI and ended up something like this:

![Screen Shot 2022-02-21 at 16 59 12](https://user-images.githubusercontent.com/16493751/154969311-38d7c9b1-641d-414e-8556-fc66a9a97082.png)

```
An unexpected error happened
TypeError: t.id is undefined
```

Now I have to figure out how can I restore the previous version of my dashboard. Since I'm not familiar with Grafana API, I probably going to find a cURL that dumps my entire dashboard and after making some fixups, I'll send a POST request with a new JSON. Which would be a nicer process such as:

* If given JSON is incorrect, consider show a warning
* If I got an error just after new JSON save, there should be an option that says: `Rollback to previous state`
* It would be nice to have a JSON editor that allows us to re-configure the given incorrect data.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
